depal_fj.py

```diff
@@ -113,152 +101,157 @@ def list_boundary_types():
     data = list(itertools.filterfalse(lambda x: x == "", data))
     return pd.DataFrame(data)
 
+
 # AOI from GeoJson File (use geojson.io)
 def get_area_from_geojson(geojson_file):
     local = gpd.read_file(geojson_file)
     area_of_interest = local
     return area_of_interest
 
+
 # AOI from a Country Nation Boundary
 def get_national_boundary():
     cadm = gdf_fj_adm[gdf_fj_adm["country"] == country]
     return cadm.dissolve().geometry
 
-#BBOX from Area Of Interest (AOI)
+
+# BBOX from Area Of Interest (AOI)
 def get_bbox(aoi):
     bbox = rasterio.features.bounds(aoi)
     return bbox
 
-#indicies
-def get_annual_rgb(aoi, year):
-    bands=["B04", "B03", "B02"]
-    bbox = rasterio.features.bounds(aoi)
+
+def get_annual_rgb(aoi: gpd.GeoSeries, year: str) -> xr.Dataset:
+    # Use the geometry in the search and load functions
+    geometry = aoi.geometry[0]
+    bands = ["B04", "B03", "B02"]
     client = Client.open(DEP_CATALOG)
-    items = client.search(
-        collections=["dep_s2_geomad"],
-        bbox=bbox,
-        datetime=str(year)
-    ).items()
-    data = load(items, bands=bands, chunks=chunks, bbox=bbox, resolution=10).squeeze("time")
+
+    # Search for items
+    items = list(
+        client.search(
+            collections=["dep_s2_geomad"], intersects=geometry, datetime=str(year)
+        ).items()
+    )
+
+    # Load those items
+    data = load(
+        items, bands=bands, chunks=chunks, geopolygon=geometry, resolution=100
+    ).squeeze("time")
+
     return data
```

Comment:
This is the only meaningful change... sorry about all the other formatting fixes!